board/stm32h7/peripherals.h

```diff
@@ -0,0 +1,299 @@
+void gpio_usb_init(void) {
+  // A11,A12: USB:
+  set_gpio_alternate(GPIOA, 11, GPIO_AF10_OTG1_FS);
+  set_gpio_alternate(GPIOA, 12, GPIO_AF10_OTG1_FS);
+  GPIOA->OSPEEDR = GPIO_OSPEEDR_OSPEED11 | GPIO_OSPEEDR_OSPEED12;
+}
+
+void gpio_usart2_init(void) {
+  // A2,A3: USART 2 for debugging
+  set_gpio_alternate(GPIOA, 2, GPIO_AF7_USART2);
+  set_gpio_alternate(GPIOA, 3, GPIO_AF7_USART2);
+}
+
+// Common GPIO initialization
+void common_init_gpio(void){
+  // FIXME: as soon as development is over - clean up this section and turn off unneeded IOs with register_set
+  /// E2,E3,E4: RGB LED
+  set_gpio_output_type(GPIOE, 2, OUTPUT_TYPE_PUSH_PULL);
```

Comment:
this needs a bunch of cleanup. Why do you even set these properties one at a time? And why not only when you use them? For the F4 we just have something like `set_gpio_output` which takes care of all the initialization as well